app/models/game.rb

```diff
@@ -8,4 +8,30 @@ class Game < ApplicationRecord
           OR
           (white_player_id IS NOT NULL AND black_player_id IS NULL)')
         }
+
+  class Board
+    attr_reader :grid
+    
+    def initialize(pieces)
+      @pieces = pieces
+      @grid = Array.new(8) { Array.new(8) }
+      fill_grid
+    end
+    
+    private
+    
+    def pieces
```

Comment:
I read from an online source that Ruby gives out a warning when those shortcuts are private. But it was written in 2014 and seems to be outdated; I just tested it out and there are no warnings.